editoast/src/views/search.rs

```diff
@@ -683,6 +685,47 @@ pub(super) struct SearchResultItemScenario {
     tags: Vec<String>,
 }
 
+#[derive(Search, Serialize, ToSchema)]
+#[search(
+    table = "train_schedule",
+    column(name = "timetable_id", data_type = "integer"),
+    column(name = "train_name", data_type = "string")
+)]
+#[allow(unused)]
+/// A search result item for a query with `object = "trainschedule"`
+pub(super) struct SearchResultItemTrainSchedule {
+    #[search(sql = "train_schedule.id")]
+    id: u64,
+    #[search(sql = "train_schedule.train_name")]
+    train_name: String, // useless puisqu'on en a besoin pour la request?
+    #[search(sql = "train_schedule.labels")]
+    labels: Vec<String>,
+    #[search(sql = "train_schedule.rolling_stock_name")]
+    rolling_stock_name: String,
+    #[search(sql = "train_schedule.timetable_id")]
+    timetable_id: u64, // useless puisqu'on en a besoin pour la request?
+    #[search(sql = "train_schedule.start_time")]
+    start_time: String,
+    #[search(sql = "train_schedule.schedule")]
+    schedule: Vec<String>,
+    #[search(sql = "train_schedule.margins")]
+    margins: Vec<String>,
+    #[search(sql = "train_schedule.initial_speed")]
+    initial_speed: f64,
+    #[search(sql = "train_schedule.comfort")]
+    comfort: i16,
+    #[search(sql = "train_schedule.path")]
+    path: Vec<String>,
+    #[search(sql = "train_schedule.constraint_distribution")]
+    constraint_distribution: i16,
+    #[search(sql = "train_schedule.speed_limit_tag")]
+    speed_limit_tag: Vec<String>,
+    #[search(sql = "train_schedule.power_restrictions")]
+    power_restrictions: Vec<String>,
+    #[search(sql = "train_schedule.options")]
+    options: Vec<String>,
```

Comment:
Many of these `Vec<String>` aren't actually string arrays.

* `margins`
* `options`
* `path`
* `power_restrictions`
* `schedule`

You can find the actual type they deserialize to in `models::train_schedule::TrainSchedule`.

Reply:
Thank you, I updated the code according to suggestions